src/compiler/program.ts

```diff
@@ -2670,7 +2670,7 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             return flatDiagnostics;
         }
 
-        const { diagnostics, directives } = getDiagnosticsWithPrecedingDirectives(sourceFile, sourceFile.commentDirectives, flatDiagnostics);
+        const { diagnostics, directives } = getDiagnosticsWithPrecedingDirectives(sourceFile, sourceFile.commentDirectives, flatDiagnostics, programDiagnostics.getDiagnostics(sourceFile.fileName));

```

Comment:
Dont think we should be doing this as part of merged bind and check diagnostics. incremental/watch mode caches these diagnostics and reuses them without having to call this but program diagnostics can change). I think the error issuing needs to happen in `getSemanticDiagnostics` instead.

Reply:
Sorry for the delay @sheetalkamat - I've re-read this a few times and I'm not sure I see a best way to resolve. Could you give me a few pointers please?

I think knowing these bits of info would be helpful:

* _"incremental/watch mode caches these diagnostics and reuses them"_: where does this happen? Are you referring to `cachedBindAndCheckDiagnosticsForFile` & similar caches passed as `cache` to `getAndCacheDiagnostics`
* _program diagnostics can change_: does this mean the cache for a file should consider its diagnostics? I don't see a good way to make it so `programDiagnostics.getDiagnostics(sourceFile.fileName)` becomes a part of the cache for a file

Thanks!